sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala

```diff
@@ -176,6 +178,28 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSQLContext {
     }
   }
 
+  test("read standard Parquet file under legacy mode") {
+    withTempPath { dir =>
+      val path = dir.getCanonicalPath
+
+      withSQLConf(SQLConf.PARQUET_WRITE_LEGACY_FORMAT.key -> "false") {
+        sqlContext
+          .range(4)
+          .selectExpr("NAMED_STRUCT('a', id, 'b', ARRAY(id, id + 1, id + 2)) AS s")
+          .write
+          .parquet(path)
+      }
+
+      withSQLConf(SQLConf.PARQUET_WRITE_LEGACY_FORMAT.key -> "true") {
```

Comment:
This config will not affect the reading path, what's the purpose of this test? I think it's already covered by `testStandardAndLegacyModes`